### PR TITLE
[5.2] Add the opposite method of isDirty, the isPristine

### PR DESCRIPTION
**Pull request by pespantelis**

### CODE REVIEW

**Comment by mnabialek** [comment]
@pespantelis  Does it make sense? You can use `!isDirty` and in case you would like to implement new method as opposite I would rather use return `!$this->isDirty()` implementation for this

**Comment by pespantelis** [comment]
@mnabialek yeah sure I can use the `!$this->isDirty()`, but it's cleaner to use the `$this->isPristine` IMO.

Moreover, Laravel has accustomed us to such facilities!
[`fails()` vs `passes()`](https://github.com/laravel/framework/blob/5.3/src/Illuminate/Validation/Validator.php#L474-L477)
[`shouldReport()` vs `shouldntReport()`](https://github.com/laravel/framework/blob/5.3/src/Illuminate/Foundation/Exceptions/Handler.php#L78-L81)
[`cant()` vs `can()`](https://github.com/laravel/framework/blob/5.3/src/Illuminate/Foundation/Auth/Access/Authorizable.php#L28-L31)
[`denies()` vs `allows()`](https://github.com/laravel/framework/blob/5.3/src/Illuminate/Auth/Access/Gate.php#L184-L187)

**Comment by mnabialek** [comment]
@pespantelis Yes, but if you take a look at implementation for example `fails()` implementation looks like this: `return ! $this->passes();` - it does not repeat the whole implementation that is included in `passes` method and this is what exactly you are doing in your implementation

**Comment by pespantelis** [comment]
@mnabialek please check my edited comment.

**Comment by GrahamCampbell** [comment]
There is too much repeated code here.

**Comment by Daursu** [comment]
Maybe changing the code to:
```php
public function isPristine($attributes = null)
{
      return ! $this->isDirty($attributes);
}
```

might be cleaner and reduces duplication.

**Comment by pespantelis** [comment]
Ok, now I'm using the `splat operator`. I have added tests too.

**Comment by pespantelis** [comment]
Hmm.. maybe I should make a new PR to the 5.3 branch, because the `splat operator` is introduced in `php 5.6`.

@GrahamCampbell right?

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -3162,23 +3162,7 @@ public function isDirty($attributes = null)
...
-        }
-
-        foreach ($attributes as $attribute) {
-            if (array_key_exists($attribute, $dirty)) {
-                return false;
-            }
-        }
-
-        return true;
+        return ! $this->isDirty(...func_get_args());
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: Thanks, but we can't use this unpacking on L5.2 because we need to support PHP 5.5.

Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: Just saw your other comment. Yeh L5.3 would be fine, though, this is still possible here with different syntax.:)

**Comment by Daursu** [comment]
Just pass $attributes to the `isDirty` function. You don't the the splat here, as all the unpacking is taking care of in the `isDirty` function.

File: src/Illuminate/Database/Eloquent/Model.php (hunk 1 above)
Comment (pespantelis) [inline src/Illuminate/Database/Eloquent/Model.php]: Yeah, with the `call_user_func_array`. Should I change it or make a new PR to 5.3?

**Comment by pespantelis** [comment]
@Daursu if I pass the `$attributes` param, the below does not work:
```php
$model->isPristine('foo', 'bar');
```

**Comment by Daursu** [comment]
@pespantelis you're right. In that case you can use
```php
call_user_func_array(array($this, "isDirty"), func_get_args());
```

**Comment by taylorotwell** [comment]
Creative method naming :smile:

**Comment by taylorotwell** [comment]
Can you send this to 5.3 with `isClean` as the method name?

**Pull request closed without merging**